portal/helpers/email.py

```diff
@@ -86,24 +89,47 @@ def send_email(sender, recipients, subject, text_content, html_content=None,
     message.send()
 
 
-def send_verification_email(request, userProfile, new_email=None):
-    verification = EmailVerification.objects.create(
-        user=userProfile,
-        email=new_email,
+def generate_token(user, email="", preverified=False):
+    return EmailVerification.objects.create(
+        user=user.userprofile,
+        new_user=user,
+        email=email,
         token=uuid4().hex[:30],
-        expiry=timezone.now() + timedelta(hours=1))
+        expiry=timezone.now() + timedelta(hours=1),
+        verified=preverified
```

Comment:
Change the method for signing a user up directly, e.g. in tests, so duplicate code is reduced.